### PR TITLE
chore: update undici to latest

### PR DESCRIPTION
**Pull request by petebacondarwin**
Also, aligns the names of workspaces in the monorepo with their containing folder, to help keep things consistent.

Note the HeadersInit type changed slightly so it needed a small refactor. See https://github.com/nodejs/undici/pull/1199

### CODE REVIEW

File: packages/wrangler/src/cfetch/internal.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ function truncate(text: string, maxLength: number): string {
 function cloneHeaders(
   headers: HeadersInit | undefined
 ): Record<string, string> {
-  return { ...headers };
+  return Array.isArray(headers) ? Object.fromEntries(headers) : { ...headers };
```
Comment (GregBrimble) [inline packages/wrangler/src/cfetch/internal.ts:77]: `headers: HeadersInit` could also be a `Headers` instance which won't properly be spread by `{ ...headers }`

**Comment by threepointone** [comment]
Oh thank god they fixed it, the Iterable signature was confusing as hell to manage. Thanks for updating it here!

**Review by GregBrimble — APPROVED** [review]

**Review by threepointone — APPROVED** [review]

File: packages/wrangler/src/cfetch/internal.ts (hunk 1 above)
Comment (petebacondarwin) [reply packages/wrangler/src/cfetch/internal.ts:77]: In our code, we never actually pass in a `Headers` object, so we kinda get away with this.
But I will make a better fix.

**Comment by threepointone** [comment]
we should also update this on miniflare so we don't ship two versions

**Comment by petebacondarwin** [comment]
> we should also update this on miniflare so we don't ship two versions

See https://github.com/cloudflare/miniflare/issues/220

**Pull request merged**